### ISSUE TITLE
MedaManage VideoView cannot play https video stream

### ISSUE THREAD

**Issue by Reza-Mortazavi** [martijn00/XamarinMediaManager#317]
I use shared project with Xamarin Forms 2.5.0.91635 in VS 2017 and installed Plugin.MediaManager and Plugin.MediaManager.Forms both 0.4.5 and video is playing on a modal page on a Android 4.4 device.
min Android version 4.1 compiled by Android 8

on xaml page:

    <forms:VideoView x:Name="VideoView" HorizontalOptions="FillAndExpand" 
        VerticalOptions="FillAndExpand" />

on cs file:

    public string VideoUrl { set ; private get; }
    private IPlaybackController PlaybackController => CrossMediaManager.Current.PlaybackController;
    private async void M_Player_OnAppearing(object sender, EventArgs e)
    {
        VideoView.AspectMode = VideoAspectMode.AspectFit;
        //VideoView.Source = VideoUrl;
        //await PlaybackController.Play();
        await CrossMediaManager.Current.Play(VideoUrl, MediaFileType.Video, ResourceAvailability.Remote);
    }

for ` http://domainname/getvideo/movie1/webm ` code works perfect
for ` https://domainname/getvideo/movie1/webm ` modal page remains black with no exception and play fails.

both urls show up on browser without problem.

this is debug output (trimmed to connection error):

> 11-23 05:04:32.463 E/MediaPlayer(21197): setDataSource: IOException! uri=https://domainname /getvideo/movie1/webm/
> 11-23 05:04:32.463 D/MediaPlayer(21197): Couldn't open file on client side, trying server side
> 11-23 05:04:32.463 D/MediatekClassFactory(21197): createInstance(): Begin = 668976713
> 11-23 05:04:32.463 D/MediatekClassFactory(21197): create Instance with :  interface com.mediatek.common.media.IOmaSettingHelper
> 11-23 05:04:32.463 W/MediatekClassFactory(21197): OP not exist!, Get obj from default class
> 11-23 05:04:32.464 D/MediatekClassFactory(21197): create Instance from operator library :  com.mediatek.op.media.DefaultOmaSettingHelper
> 11-23 05:04:32.464 D/MediatekClassFactory(21197): createInstance(): End = 668976713
> 11-23 05:04:32.464 V/DefaultOmaSettingHelper(21197): isOMAEnabled: enabled=true
> 11-23 05:04:32.464 V/DefaultOmaSettingHelper(21197): judgeStreamingType(https://domainname/getvideo/movie1/webm/)
> 11-23 05:04:32.464 V/DefaultOmaSettingHelper(21197): judgeStreamingType: type=0
> 11-23 05:04:32.464 V/DefaultOmaSettingHelper(21197): setSettingHeader: headers unchanged. uri=https://domainname/getvideo/movie1/webm/
> 11-23 05:04:32.467 D/MediaPlayer(21197): Don't notify duration to com.companyname.DataHttpClient!
> 11-23 05:04:32.467 I/SurfaceView(21197): surfaceCreated callback -
> 11-23 05:04:32.467 I/SurfaceView(21197): surfaceChanged -- format=4 w=1280 h=577
> 11-23 05:04:32.467 I/SurfaceView(21197): surfaceChanged callback +
> 11-23 05:04:32.467 I/SurfaceView(21197): surfaceChanged callback -
> 11-23 05:04:32.467 I/SurfaceView(21197): surfaceRedrawNeeded
> 11-23 05:04:32.467 I/SurfaceView(21197): finishedDrawing

**Comment by mike-rowley** [comment]
Does the problem exist on newer versions of Android, like 7 or 8?
What is serving up the video?  Is it direct from a hardware device?  Or a file on a web server?

**Comment by Reza-Mortazavi** [comment]
Unfortunately I don't have Android 7/8 for the time being and my emulator does not play videos so I have to check it on a device.
Videos are Web API 2 streams. tested both locally and from web.

available samples:

working
```
http://webapi.evrinsoft.com/getvideo/movie1/webm/
http://webapi.evrinsoft.com/getvideo/movie2/webm/
http://webapi.evrinsoft.com/getvideo/movie1/mp4/
http://webapi.evrinsoft.com/getvideo/movie2/mp4/
```

not working
```
https://webapi.evrinsoft.com/getvideo/movie1/webm/
https://webapi.evrinsoft.com/getvideo/movie2/webm/
https://webapi.evrinsoft.com/getvideo/movie1/mp4/
https://webapi.evrinsoft.com/getvideo/movie2/mp4/
```

I appreciate your time

**Comment by mike-rowley** [comment]
I HIGHLY suggest getting a new device, testing software on 4 year old Device/OS especially in the mobile space is a recipe for a release disaster.  I've seen less issues with backwards compatibility than forward compatibility.

I have tested your HTTPS link in the Forms Sample app and it works on 8.0 and 5.0.  I also did some research into the actual error you are getting (VideoView(13411): Error: 1,-1004) and there is a mountain of information about HLS nightmares on Android 4.0-4.4.  From what I read it looks like there were some serious issues around streaming on 4.  If you find the resolution feel free to submit a PR, we'd be more than happy to have a look at it.

**Comment by Reza-Mortazavi** [comment]
I understand the new device concept but Android 4 devices are still popular. For example I personally have had no reason to change my handy 4.4 Yoga tablet!

I spent three whole days on this problem, reading and testing everything related to this issue even reviewing your source code, and when I gave up came here and wrote. I just opened the project and run it and guess what... all videos played on same settings as before. I'm still unaware of any probable cause.

You gave me a good clue: as http works and https used to fail with IO error, maybe an old OS/device may have problems with network security layer or Web API security / authentication on server side. 

This is my my first time working with video streams on Xamarin/Android and still in R&D phase and I have lots to study and learn. I assume this was a temporary problem and will try to reproduce it in coming days and inform you about the results.

I appreciate your kind attention, time and response.
